snapcraft/cli/lifecycle.py

```diff
@@ -31,6 +40,9 @@ def _execute(command, parts, **kwargs):
     else:
         try:
             lifecycle.execute(command, project_options, parts)
+        except errors.SnapcraftUserPermissionError as user_error:
+            if user_error.user == 'root':
+                _reexec_as_root()
```

Comment:
exactly, build should be fine with the above "hack" to run all the live-build steps with sudo, when we tried it with the already deployed change in LP (before it got rolled back) only prime failed (note though that mksquashfs uses special options for os snaps and since we failed in prime i dont know if there would have been any subsequent errors in the "snap" step or if the filesystem permissions in teh resulting snap would have been broken)